### ISSUE TITLE
[CLI]: `pdb.set_trace()` does not surface with `wandb agent` in sweep

### ISSUE THREAD

**Issue by ahalev** [wandb/wandb#6179]
### Describe the bug

Setting a pdb trace in a script causes the script causes the script to freeze at the trace point. However, the pdb interface does not appear and the user's only option is to kill the training run.

Replace [lines 187-189](https://github.com/wandb/examples/blob/master/examples/pytorch/pytorch-cnn-fashion/train.py#L187-L188) with 

```python
if __name__ == '__main__':
   print('BEFORE TRACE')  # this prints
   import pdb; pdb.set_trace()
   print('AFTER TRACE')  # this does not
   main()
```

Steps to reproduce:

Navigate to `pytorch-cnn-fashion`
```shell
wandb sweep sweep-grid-hyperband.yaml
wandb agent <SWEEP_NAME>
```

### Environment

WandB version: 0.15.5

OS: MacOS 12.6

Python version: 3.8.17

Versions of relevant libraries:

**Comment by luisbergua** [comment]
Hi @ahalev, thanks you so much for reporting this and for sharing detailed steps to reproduce this issue. I see the same behavior so will raise this internally as a bug to our eng team and we'll keep you posted about the status of it! You can run the sweep using the SDK and pdb will work as expected like:

```
import wandb

def train():
    print('BEFORE TRACE')  
    import pdb; pdb.set_trace()  
    print('AFTER TRACE') 

    run = wandb.init()
    ...
    run.finish()

sweep_config = {...}

sweep_id = wandb.sweep(sweep_config)

wandb.agent(sweep_id, function=train)
```

**Comment by sydholl** [comment]
Sydney Hollingsworth commented: 
Auto-assigned

**Comment by ahalev** [comment]
Maybe I'm missing something but I'm not sure how hyperparameters are passed to the `function` used in wandb.agent. 

It's called here as `self._function`: https://github.com/wandb/wandb/blob/b1295d243b3e039c951f58134f5a9ca4f6751aed/wandb/agents/pyagent.py#L294-L299

but how is `job.config` surfaced to the function? Are they mocking `sys.argv` or something? It doesn't seem like it; it seems like you have to access `wandb.config` in the training function to get access to those variables, which is functionally different from calling a program script as `sweep` then passes command line arguments.

**Comment by luisbergua** [comment]
Hey @ahalev, thanks for your question! As you can see in [this](https://docs.wandb.ai/guides/sweeps/add-w-and-b-to-your-code) page from our docs, the hyperparameters from config are set in the `wandb.config` object, and you typically access them inside your training function using `wandb.config`

![image](https://github.com/wandb/wandb/assets/113048227/110a1a4a-8b8f-423a-8042-ec3945458de3)

**Comment by ahalev** [comment]
Right, so this method of using an agent via the sdk is fundamentally different than using it via the cli. In my case, I'm using a separate package to parse command line args; I'd have to modify it to read `wandb.config`, so this switch from cli to sdk isn't as simple as it would appear.

**Comment by luisbergua** [comment]
Thanks for your answer @ahalev! You're right that this can be done differently depending if you're using the SDK of the CLI but the recommended practice in both cases is to use the `config` object of the run (see [this](https://docs.wandb.ai/guides/sweeps/add-w-and-b-to-your-code#training-script-with-wb-python-sdk) specific section with an example of both cases) as this allows you to always access values from that config. Please let me know if that helps!

**Comment by sydholl** [comment]
WandB Internal User commented: 
luisbergua commented: 
Thanks for your answer @ahalev! You're right that this can be done differently depending if you're using the SDK of the CLI but the recommended practice in both cases is to use the `config` object of the run (see [this](https://docs.wandb.ai/guides/sweeps/add-w-and-b-to-your-code#training-script-with-wb-python-sdk) specific section with an example of both cases) as this allows you to always access values from that config. Please let me know if that helps!

**Comment by sydholl** [comment]
Luis Bergua commented: 
Hi there, I wanted to follow up on this request. Please let us know if we can be of further assistance or if your issue has been resolved.

**Comment by ahalev** [comment]
Has the original bug been fixed?

**Comment by luisbergua** [comment]
Hey @ahalev! Not yet although our Engineering Team is aware of that one so will let you know once there're any relevant updates
